readthedocs/oauth/services/github.py

```diff
@@ -456,7 +456,7 @@ def send_build_status(self, build, commit, state, link_to_build=False):
         target_url = build.get_full_url()
         statuses_url = f'https://api.github.com/repos/{owner}/{repo}/statuses/{commit}'
 
-        if not link_to_build and state == BUILD_STATUS_SUCCESS:
+        if not link_to_build and not build.version.built and not build.version.uploaded:
```

Comment:
@humitos I wonder if this is actually the proper fix after all? Have it match the logic in `get_absolute_url`?

Reply:
Hrm, I'm not 100% sure, but I don't think so. `.built` and `.uploaded` are not updated to `False` when a new build is triggered for the same version; so we will be reading an old value here.

IMO, we need this logic:

- Successful build: link to documentation
- Failed build: link to build details' page

As you may noted, the link to send does not depends on the "Commit build status" we sent to GitHub (`state == BUILD_STATUS_SUCCESS`, in that code), but it depends on the "status" of the build itself.

Actually, I'd completely remove `link_to_build` and just use the build's status inside this method. That way, we manage the logic just in one place and we don't have to worry about _where_ and _why_ we are passing `link_to_build=False/True` in different places of the code.